tests/sqlparser_common.rs

```diff
@@ -2153,6 +2153,17 @@ fn parse_create_table_as() {
     }
 }
 
+#[test]
+fn parse_create_table_as_table() {
+    let sql1 = "CREATE TABLE new_table AS TABLE old_table";
```

Comment:
In general I think we try to have the sqlparser crate parse into an AST that reflects the parsed SQL -- so rather that assuming some sematics (like `CREATE TABLE x AS y` means the same as `CREATE TABLE AS SELECT * ..` it would be more consistent to parse the create table differently.

Does that make sense?

Reply:
Yes, I can edit accordingly. Thanks!